js/modules/k6/http/request.go

```diff
@@ -177,7 +177,16 @@ func (h *HTTP) parseRequest(
 		if !requestContainsFile(data) {
 			bodyQuery := make(url.Values, len(data))
 			for k, v := range data {
-				bodyQuery.Set(k, formatFormVal(v))
+				switch reflect.TypeOf(v).Kind() { //nolint:exhaustive
+				// handle json arrays in params
+				case reflect.Slice, reflect.Array:
+					s := reflect.ValueOf(v)
+					for i := 0; i < s.Len(); i++ {
+						bodyQuery.Add(k, formatFormVal(s.Index(i)))
+					}
```

Comment:
Using a simple type switch and matching just `[]interface{}` works just as well. I have no idea how to make a js code that will give me a go array and apparently the test below doesn't do it ... so I am kind of for ditching the `reflect` ;)

Reply:
that was my objection in the original PR, but then I got this response and didn't verify for myself: https://github.com/k6io/k6/pull/1754#issuecomment-738773727

Reply:
Well I did write the code .. and we do assert to `[]interface{}` in the batch parsing code:
https://github.com/k6io/k6/blob/b9a29eb7fc75b5d903f0c9e3a307ed80d06dacc5/js/modules/k6/http/request.go#L468-L474

Reply:
Yeah, I assumed that comment to be true and didn't try a type assertion. Seems to work fine, see 0d48122b.